### DISCUSSION TITLE
how do we exclude Vue CSS `<style>` from coverage?

### DISCUSSION THREAD

**Discussion by ghiscoding** [vitest-dev/vitest#3208] [Q&A] [answered]
I'm slowly adding unit tests to my Vue 3 project and I haven't added coverage for any components yet (I'm focusing on Services/Utils first) and I was amazed by the number of lines that C8 was covering. After looking a little deeper, I see that all CSS `<style>` are actually detected as coverage, is there a way to remove these lines from being covered? I know that if I add a simple unit test to my Component, it will then cover CSS but still I'm not interested at all to see CSS in my coverage... 

So how can I exclude CSS `<style>` from showing in my coverage? 
I also found this Stack Overflow [Question](https://stackoverflow.com/questions/73097701/ignore-style-tag-on-vitest-coverage) which is asking the same

I tried `css: false` in Vitest.config, but I think that flag is meant for something else. 

![image](https://user-images.githubusercontent.com/643976/232888686-75a2362b-f81c-4376-8d92-b6fafe529798.png)

I know that I could use `/* c8 ignore start */` and stop but I'm trying to avoid having to add this to all my components

**Comment by AriPerkkio** [comment, marked as answer]
Is this file included in coverage report because of `coverage.all: true` option? Meaning that there is no test case that loads it? If yes, then `c8` will simply mark whole file as uncovered. It does not understand what code the file contains. 

You may have better luck with `provider: istanbul` which does AST based code coverage.

**Reply by ghiscoding** [reply]
It's covered because I added the `.vue` extension to be covered and since Vue puts everything in 1 file (TS/HTML/CSS), I was looking for a way to exclude CSS section from the coverage. It does indeed seem like Istanbul knows the difference and does a better job. Thanks
